src/core/lib/promise/mpsc.h

```diff
@@ -76,14 +76,19 @@ class Center : public RefCounted<Center<T>> {
     return true;
   }
 
-  // Returns the batch number that the item was sent in, or kClosedBatch if the
-  // pipe is closed.
-  uint64_t Send(T t) {
+  // Return value:
+  //  - if the pipe is closed, returns kClosedBatch
+  //  - if await_receipt is false, returns the batch number the item was sent
+  //  in.
+  //  - if await_receipt is true, returns the first sending batch number that
+  //  guarantees the item has been received.
+  uint64_t Send(T t, bool await_receipt) {
     ReleasableMutexLock lock(&mu_);
     if (batch_ == kClosedBatch) return kClosedBatch;
     queue_.push_back(std::move(t));
     auto receive_waker = std::move(receive_waker_);
-    const uint64_t batch = queue_.size() <= max_queued_ ? batch_ : batch_ + 1;
+    const uint64_t batch =
+        (!await_receipt && queue_.size() <= max_queued_) ? batch_ : batch_ + 1;
```

Comment:
Based on the comment "Every time we give queue_ to the receiver, we increment batch_", is my understanding correct that we will only receive (batch_+1) when batch_ is guaranteed to be received? Therefore we can use (batch_ + 1) as a strong signal? Thanks!